sql/_project_record_ids.sql

```diff
@@ -27,26 +27,31 @@ dbscan AS (
 project_record_join AS (
 	SELECT 
 	    a.record_id,
-	    COUNT(record_id) OVER(PARTITION BY id) as records_in_project,
+	    COUNT(record_id) OVER(PARTITION BY id) AS records_in_project,
 	    a.id,
-	    (a.geom IS NULL)::integer as no_geom,
+	    (a.geom IS NULL)::integer AS no_geom,
 	    a.geom
 	FROM dbscan a
 ),
 all_intersections AS (
-	SELECT 
-		ST_UNION(ST_INTERSECTION(a.geom, b.geom)) as intersect_geom
+	SELECT
+		ST_UNION(ST_INTERSECTION(a.geom, b.geom)) AS intersect_geom,
+		a.id
 	FROM  project_record_join a, project_record_join b
 	WHERE a.record_id < b.record_id
 	AND a.records_in_project > 1
 	AND b.records_in_project > 1
 	AND a.no_geom = 0
 	AND b.no_geom = 0
+	AND a.id = b.id
+	GROUP BY a.id
 )
 SELECT
-	array_agg(a.record_id) as project_record_ids
+	array_agg(a.record_id) AS project_record_ids
 INTO _project_record_ids
 FROM project_record_join a, all_intersections b
-WHERE ST_OVERLAPS(a.geom, b.intersect_geom)
+WHERE (ST_Overlaps(a.geom, b.intersect_geom) OR ST_Contains(a.geom, b.intersect_geom) OR ST_Within(a.geom, b.intersect_geom))
```

Comment:
To properly identify different agency projects (polygon) as clusters, we utilize three postgis functions to cover different type of edge cases:
1. `St.Overlaps` - "Two geometries overlap if they have the same dimension, each has at least one point not shared by the other (or equivalently neither covers the other), and the intersection of their interiors has the same dimension" For our use case, `st.overlaps` is meant to include polygons in which their permiter crosses over into another polygon
![st_overlaps03](https://user-images.githubusercontent.com/41444592/204376274-5161b0c1-5e7e-4c07-a60c-8647b544a2b9.png)

2. `St_Contains `- "Returns TRUE if geometry B is completely inside geometry A. A contains B if and only if no points of B lie in the exterior of A, and at least one point of the interior of B lies in the interior of A. A subtlety of the definition is that a geometry does not contain things in its boundary. Thus polygons and lines do not contain lines and points lying in their boundary." For KPDB, we want to capture agency projects (polygons) that boundary is fully "contained" within another projects as a cluster. 
![st_contains04](https://user-images.githubusercontent.com/41444592/204380095-72c9861b-488b-4be4-bce8-0ede8ca05503.png)
3. `ST_WIthin` - "Returns TRUE if geometry A is completely inside geometry B. For this function to make sense, the source geometries must both be of the same coordinate projection, having the same SRID. It is a given that if ST_Within(A,B) is true and ST_Within(B,A) is true, then the two geometries are considered spatially equal. ST_Within is the inverse of [ST_Contains](https://postgis.net/docs/ST_Contains.html). So, ST_Within(A,B) = ST_Contains(B,A)." For KPDB, we want to capture records that have the exact same geometry which is why we apply the inverse logic to st_contains(a.geom, b.intersect_geom) which would be st_contains(b.intersect_geom, a.geom)

Reply:
I think flushing out the comment for how we apply the spatial logic will be enough to approve this PR